iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -2216,7 +2216,7 @@ + (BOOL)receiveRemoteNotification:(UIApplication*)application UserInfo:(NSDictio
     else if (application.applicationState == UIApplicationStateActive) {
         [OneSignalHelper lastMessageReceived:userInfo];
 
-        if (![OneSignalHelper isRemoteSilentNotification:userInfo]) {
+        if (![OneSignalHelper isRemoteSilentNotification:userInfo] && (userInfo[@"title"] && [userInfo[@"title"] length] > 0)) {
```

Comment:
Instead of adding an extra "title" check to this if statement thinking we should add a helper that better fits what we need to check here.

Suggested new method called `isDisplayableNotification, that will check if the notification would be displayed in the Notification Center on the device. The only requirement for this is that the notification have a message body I believe. Not sure if a notification would be displayed with only a title.

Also check may not be needed at all if we can confirm `UNNotificationCenterDelegate`'s  `userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` does not fire for silent notifications. This code path might just be called from the AppDelegate which we may not need to process for iOS 10+.

Reply:
The notification is displayed with only a title and it is also displayed with only a message so I think we need to check both cases. I think the isDisplayableNotification is the way to go